sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java

```diff
@@ -254,52 +238,44 @@ private Transforms() {
                 break;
             case DATE:
-                value = new FormField(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
-                    key, new com.azure.ai.formrecognizer.models.FieldValue(FieldValueType.DATE)
-                    .setFormFieldDate(fieldValue.getValueDate()), valueText);
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
+                    key,  fieldValue.getValueDate(), valueText, FieldValueType.DATE);
                 break;
             case INTEGER:
-                value = new FormField(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
-                    key, new com.azure.ai.formrecognizer.models.FieldValue(FieldValueType.INTEGER)
-                    .setFormFieldInteger(fieldValue.getValueInteger()), valueText);
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
+                    key, fieldValue.getValueInteger(), valueText, FieldValueType.INTEGER);
                 break;
             case NUMBER:
-                value = new FormField(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
-                    key, new com.azure.ai.formrecognizer.models.FieldValue(FieldValueType.FLOAT)
-                    .setFormFieldFloat(fieldValue.getValueNumber()), valueText);
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
+                    key, fieldValue.getValueNumber(), valueText, FieldValueType.FLOAT);
                 break;
             case ARRAY:
-                value = new FormField(setDefaultConfidenceValue(fieldValue.getConfidence()), null, key,
-                    new com.azure.ai.formrecognizer.models.FieldValue(FieldValueType.LIST)
-                        .setFormFieldList(toFormFieldArray(fieldValue.getValueArray(), readResults)), null);
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), null, key,
+                     toFormFieldArray(fieldValue.getValueArray(), readResults), null, FieldValueType.LIST);
                 break;
             case OBJECT:
-                value = new FormField(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
-                    key, new com.azure.ai.formrecognizer.models.FieldValue(FieldValueType.MAP)
-                    .setFormFieldMap(toFormFieldObject(fieldValue.getValueObject(), pageNumber, readResults)), valueText
-                );
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
+                    key, toFormFieldObject(fieldValue.getValueObject(), pageNumber, readResults), valueText,
+                    FieldValueType.MAP);
                 break;
             default:
                 throw LOGGER.logExceptionAsError(new RuntimeException("FieldValue Type not supported"));
```

Comment:
Since `FieldValueType` is an `ExpandableStringEnum`, should the default be to create a `FormField` with `Object` value instead of throwing an exception?

Reply:
Discussed offline, currently, a newer version of FieldValue can only be supported with a swagger update requiring service version update.